public/app/features/dashboard-scene/sharing/public-dashboards/ConfigPublicDashboard.tsx

```diff
@@ -0,0 +1,83 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { SceneComponentProps, sceneGraph } from '@grafana/scenes';
+import { useStyles2 } from '@grafana/ui';
+import { contextSrv } from 'app/core/core';
+import {
+  useDeletePublicDashboardMutation,
+  useUpdatePublicDashboardMutation,
+} from 'app/features/dashboard/api/publicDashboardApi';
+import ConfigPublicDashboardComponent from 'app/features/dashboard/components/ShareModal/SharePublicDashboard/ConfigPublicDashboard/ConfigPublicDashboard';
+import { PublicDashboard } from 'app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboardUtils';
+import { AccessControlAction } from 'app/types';
+
+import { ShareModal } from '../ShareModal';
+
+import { ConfirmModal } from './ConfirmModal';
+import { SharePublicDashboardTab } from './SharePublicDashboardTab';
+import { useUnsupportedDatasources } from './hooks';
+
+interface Props extends SceneComponentProps<SharePublicDashboardTab> {
+  publicDashboard?: PublicDashboard;
+  isGetLoading?: boolean;
+}
+
+export function ConfigPublicDashboard({ model, publicDashboard, isGetLoading }: Props) {
+  const styles = useStyles2(getStyles);
+
+  const hasWritePermissions = contextSrv.hasPermission(AccessControlAction.DashboardsPublicWrite);
+  const { dashboardRef } = model.useState();
+  const dashboard = dashboardRef.resolve();
+  const { isDirty } = dashboard.useState();
+  const [update, { isLoading: isUpdateLoading }] = useUpdatePublicDashboardMutation();
+  const [deletePublicDashboard, { isLoading: isDeleteLoading }] = useDeletePublicDashboardMutation();
+  const hasTemplateVariables = (dashboard.state.$variables?.state.variables.length ?? 0) > 0;
+  const unsupportedDataSources = useUnsupportedDatasources(dashboard);
+  const isDataLoading = isUpdateLoading || isGetLoading || isDeleteLoading;
+  const timeRangeState = sceneGraph.getTimeRange(model);
+  const timeRange = timeRangeState.useState();
+
+  return (
+    <ConfigPublicDashboardComponent
+      publicDashboard={publicDashboard}
+      unsupportedDatasources={unsupportedDataSources}
+      isLoading={isDataLoading}
+      onUpdate={(updatedPublicDashboard) => {
+        update({ dashboard, payload: updatedPublicDashboard });
+      }}
```

Comment:
Agree that update and isLoading can be inside ConfigPublicDashboardComponent, but I think publicDashboard still has to be in SharePublicDashboard since it's used for this conditional:

https://github.com/grafana/grafana/blob/d68d31c63ca80651b0fe4071bc409778cb8ecf79/public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard.tsx#L34-L37

Reply:
since the use of the RTK query hook in many components ends up returning the same retrieved data, then that hook can be used in the parent and in the child. 
I mean, it won't execute twice the API call, but it will return exactly the already fetched data (since it was first fetched by the parent) 
Here there's good info 
https://stackoverflow.com/questions/70647677/how-to-access-data-in-child-component-of-query-called-in-parent-component-in-rtk
https://redux.js.org/tutorials/essentials/part-8-rtk-query-advanced#cache-data-subscription-lifetimes